src/main/java/edu/greenblitz/utils/GBMath.java

```diff
@@ -1,20 +1,92 @@
 package edu.greenblitz.utils;
 
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Translation2d;
+
 public class GBMath {
-	/**
-	 *
-	 * modulo (2,6) -> 0
-	 * modulo (2,3) -> 1
-	 * modulo (-2,3) -> 1
-	 * modulo (3,-2) -> -1 -> 1
-	 *
-	 * @param x the number to be modulu-ed
-	 * @param y the modulo base
-	 * @return the correct modulo result
-	 */
-	public static double absoluteModulo(double x, double y) {
-		return ((x % y) + y) % y;
-	}
-	
-	
+    /**
+     * modulo (2,6) -> 0
+     * modulo (2,3) -> 1
+     * modulo (-2,3) -> 1
+     * modulo (3,-2) -> -1 -> 1
+     *
+     * @param x the number to be modulu-ed
+     * @param y the modulo base
+     * @return the correct modulo result
+     */
+    public static double absoluteModulo(double x, double y) {
+        return ((x % y) + y) % y;
+    }
+
+    /**
+     * Takes a length and angle and converts it into x and y
+     *
+     * @param length the length of the point
+     * @param angle  the angle of the point
+     * @return a Translation2d which houses the result
+     */
+    public static Translation2d convertToCartesian(double length, double angle) {
+        return new Translation2d(length * Math.cos(angle), length * Math.sin(angle));
+    }
+
+    /**
+     * Takes an x and y and converts it into polar coordinates
+     *
+     * @param x the x coordinate of the point
+     * @param y the y coordinate of the point
+     * @return a Pair which houses the result in this format (length,angle)
+     */
+    public static Pair<Double, Double> convertToPolar(double x, double y) {
+        return new Pair<>(Math.sqrt(x * x + y * y), Math.atan2(y, x));
+    }
+
+    /**
+     * Takes an x and y position and speeds, and radius of polar conversion and converts it into polar speeds
+     *
+     * @param xPos      the x coordinate of the point
+     * @param yPos      the y coordinate of the point
+     * @param xVelocity the x Velocity of the point
+     * @param yVelocity the y Velocity of the point
+     * @return a Pair which houses the result in this format (length,angle)
+     */
+    public static Pair<Double, Double> convertToPolarSpeeds(double xPos, double yPos, double xVelocity, double yVelocity) {
+
+        double distance = Math.sqrt(xPos * xPos + yPos * yPos);
+
+        double lengthVelocity = xPos * (xVelocity) + yPos * (yVelocity);
+        lengthVelocity /= distance;
+
+        double angularVelocity = xPos * yVelocity - yPos * xVelocity;
+        angularVelocity /= distance;
+        return new Pair<>(lengthVelocity, angularVelocity);
+    }
+
+    public static double distance(Translation2d p1, Translation2d p2) {
```

Comment:
p1?

The McLaren P1 produced by McLaren Automotive, a supercar with a mid-engine layout with rear-wheel drive and a roof structure safety cage concept called MonoCage?

![image](https://github.com/GreenBlitz/Charrrged-Up-2023/assets/80216549/b6694b7d-bf9a-4545-b54b-a2ac9ea9cefb)

How is that related to "new arm arm cr"? (GREAT name by the way)

Reply:
Also `distance` is a bit of a vague name

Reply:
Changed to point.
About distance the function is doing the distance equation so i think its a fair name.